huaweicloud/services/evs/resource_huaweicloud_evs_volumes_batch_expand.go

```diff
@@ -0,0 +1,145 @@
+package evs
+
+import (
+	"context"
+	"strings"
+
+	"github.com/hashicorp/go-uuid"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+
+	"github.com/chnsz/golangsdk"
+
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/config"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/utils"
+)
+
+var volumesBatchExpandNonUpdatableParams = []string{
+	"volumes",
+	"volumes.*.id",
+	"volumes.*.new_size",
+	"is_auto_pay",
+}
+
+// @API EVS POST /v5/{project_id}/volumes/batch-extend
+func ResourceVolumesBatchExpand() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceVolumesBatchExpandCreate,
+		ReadContext:   resourceVolumesBatchExpandRead,
+		UpdateContext: resourceVolumesBatchExpandUpdate,
+		DeleteContext: resourceVolumesBatchExpandDelete,
+
+		CustomizeDiff: config.FlexibleForceNew(volumesBatchExpandNonUpdatableParams),
+
+		Schema: map[string]*schema.Schema{
+			"region": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+			},
+			"volumes": {
+				Type:     schema.TypeList,
+				Required: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"id": {
+							Type:     schema.TypeString,
+							Required: true,
+						},
+						"new_size": {
+							Type:     schema.TypeInt,
+							Required: true,
+						},
+					},
+				},
+			},
+			"is_auto_pay": {
+				Type:     schema.TypeBool,
+				Optional: true,
+			},
+			"enable_force_new": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ValidateFunc: validation.StringInSlice([]string{"true", "false"}, false),
+				Description:  utils.SchemaDesc("", utils.SchemaDescInput{Internal: true}),
+			},
+		},
+	}
+}
+
+func buildVolumesBatchExpandBodyParams(d *schema.ResourceData) map[string]interface{} {
+	volumes := d.Get("volumes").([]interface{})
+	volumesList := make([]map[string]interface{}, len(volumes))
+
+	for i, v := range volumes {
+		volume := v.(map[string]interface{})
```

Comment:
when the script is configured as: valumes {nil}，code `volume := v.(map[string]interface{})` may cause convert error.
```suggestion
		volume, ok := v.(map[string]interface{})
		if !ok {
		  continue
		}
		volumesList = appeng(volumesList, map[string]interface{}{
			"id":       volume["id"],
			"new_size": volume["new_size"],
		})
```